xla/stream_executor/rocm/rocm_executor.cc

```diff
@@ -735,7 +741,11 @@ DeviceMemoryBase RocmExecutor::Allocate(uint64_t size, int64_t memory_space) {
   switch (static_cast<MemoryType>(memory_space)) {
     case MemoryType::kCollective:
     case MemoryType::kDevice:
-      return DeviceMemoryBase(DeviceAllocate(rocm_context_, size), size);
+      return DeviceMemoryBase(
+          DeviceAllocate(rocm_context_, size, /*is_fine_grained*/ true), size);
+    case MemoryType::kFineGrainedDevice:
+      return DeviceMemoryBase(
+          DeviceAllocate(rocm_context_, size, /*is_fine_grained*/ true), size);
```

Comment:
so is this `hipExtMallocWithFlags` always used by default, do we know what's trade-off on normal `hipMalloc` ?

Reply:
Sorry for the typo.

- `kDevice` --> `hipMalloc`
- `kFineGrainedDevice` --> `hipExtMallocWithFlags`

Now only the handwritten all_reduce uses this fine-grained memory and it doesn't affect other memory allocation.